src/components/ReportActionItem/MoneyRequestView.tsx

```diff
@@ -397,50 +370,42 @@ function MoneyRequestView({report, parentReport, parentReportActions, policyCate
     );
 }
 
-MoneyRequestView.propTypes = propTypes;
-MoneyRequestView.defaultProps = defaultProps;
 MoneyRequestView.displayName = 'MoneyRequestView';
 
-export default compose(
-    withCurrentUserPersonalDetails,
-    withOnyx({
-        session: {
-            key: ONYXKEYS.SESSION,
-        },
-        policy: {
-            key: ({report}) => `${ONYXKEYS.COLLECTION.POLICY}${report.policyID}`,
-        },
-        policyCategories: {
-            key: ({report}) => `${ONYXKEYS.COLLECTION.POLICY_CATEGORIES}${report.policyID}`,
-        },
-        policyTags: {
-            key: ({report}) => `${ONYXKEYS.COLLECTION.POLICY_TAGS}${report.policyID}`,
-        },
-        parentReport: {
-            key: ({report}) => `${ONYXKEYS.COLLECTION.REPORT}${report.parentReportID}`,
-        },
-        parentReportActions: {
-            key: ({report}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report ? report.parentReportID : '0'}`,
-            canEvict: false,
+export default withOnyx<MoneyRequestViewPropsWithoutTransaction, MoneyRequestViewOnyxPropsWithoutTransaction>({
+    policy: {
+        key: ({report}) => `${ONYXKEYS.COLLECTION.POLICY}${report.policyID}`,
+    },
+    policyCategories: {
+        key: ({report}) => `${ONYXKEYS.COLLECTION.POLICY_CATEGORIES}${report.policyID}`,
+    },
+    policyTags: {
+        key: ({report}) => `${ONYXKEYS.COLLECTION.POLICY_TAGS}${report.policyID}`,
+    },
+    parentReport: {
+        key: ({report}) => `${ONYXKEYS.COLLECTION.REPORT}${report.parentReportID}`,
+    },
+    parentReportActions: {
+        key: ({report}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report ? report.parentReportID : '0'}`,
+        canEvict: false,
+    },
+    transactionViolations: {
+        key: ({report}) => {
+            const parentReportAction = ReportActionsUtils.getParentReportAction(report);
+            const originalMessage = parentReportAction?.actionName === CONST.REPORT.ACTIONS.TYPE.IOU ? parentReportAction.originalMessage : undefined;
+            const transactionID = originalMessage?.IOUTransactionID ?? 0;
+            return `${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transactionID}`;
         },
-    }),
-    withOnyx({
+    },
+})(
+    withOnyx<MoneyRequestViewProps, MoneyRequestViewTransactionOnyxProps>({
```

Comment:
Why did you move `transactionViolations` and `policyTags` to the other HOC? Isn't possible to keep it here?

Reply:
It's possible, but I don't see a reason. The second HOC exists to use data from the first HOC, so transactions use `parentReportActions`, that's why `transactions` are in the second HOC, but `transactionViolations` and `policyTags` doesn't use any data from the first HOC